src/NUnitFramework/framework/Constraints/Constraint.cs

```diff
@@ -139,16 +140,16 @@ public override string ToString()
         }
 
         /// <summary>
-        /// Returns the string representation of this constraint
+        /// Returns the string representation of this constraint and the passed in arguments
         /// </summary>
-        protected virtual string GetStringRepresentation()
+        protected string GetStringRepresentation(IEnumerable arguments)
```

Comment:
question: Since `Constraint` is public, this overload technically becomes a new "public" API as a custom derived Constraint may try to call this. I usually default to `protected` in this case out of habit, but figured I'd ask: would `private protected` be wanted here instead?

Reply:
I had to look up `private protected` because it logically didn't make sense. Since when does `private` mean `internal`?
But that is because `protected internal` means either `derived (anywhere)` or `direct calls from the assembly` instead of `only derived classes inside the assembly`

I think this can be `protected` in case an external `Constraint` has a single `IEnumerable` argument which by default gets formatted as just the type name.
There are at least two existing constraints: `CollectionSubsetConstraint` and `CollectionSupersetConstraint` which could benefit from this call as their current (tested against) string representation is: `<s...setof System.Int32[]>`